lib/offsite_payments/integrations/paytm.rb

```diff
@@ -0,0 +1,475 @@
+module OffsitePayments #:nodoc:
+  module Integrations #:nodoc:
+    module Paytm
+	
+	  require 'openssl'
+	  require 'base64'
+	  require 'digest'
+	  require 'securerandom'
+	
+	def new_pg_encrypt(params)
+		if (params.class != Hash) || (params.keys == [])
+		  return false
+		end
+		if !params.has_key?(:key)
+		  return false
+		end
+		encrypted_data = Hash[]
+		key = params.delete(:key)
+		keys = params.keys
+		aes = OpenSSL::Cipher::Cipher.new("aes-128-cbc")
+		begin
+		  keys.each do |k|
+			data = params[k]
+			aes.encrypt
+			aes.key = key
+			aes.iv = '@@@@&&&&####$$$$'
```

Comment:
Using a fixed iv is dangerous, should be randomized on each encrypt and included at the beginning of the encrypted data.

https://cwe.mitre.org/data/definitions/329.html